hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleContextImpl.java

```diff
@@ -120,6 +122,48 @@ public Set<Key> requiredNonPayerKeys() {
         return Collections.unmodifiableSet(requiredNonPayerKeys);
     }
 
+    @NonNull
+    @Override
+    public Set<Key> optionalNonPayerKeys() {
+        return Collections.unmodifiableSet(optionalNonPayerKeys);
+    }
+
+    @NonNull
+    @Override
+    public PreHandleContext optionalKey(@NonNull final Key key) {
+        if (!key.equals(payerKey) && isValid(key)) {
```

Comment:
requireNonNull(key)

Reply:
Our standard allows eliding that check if the exception will be thrown in the first line (as here).  This results in a better message (the exception message will name the variable).  Using requireNonNull just forces the exception to be thrown at the beginning of the method; it's not helpful (and arguably harmful) if the code will do so without it.

Reply:
Curious @jsync-swirlds, what standard are you referring to? I wasn't aware one existed

Also, is there a technical significance to the word 'eliding' beyond its dictionary definition?

Reply:
Just dictionary definition on the meaning.  It's more a habit than anything else.
The recommendations I'm referencing stem from a discussion about using Objects.requireNonNull (somewhere in slack; I don't believe it ever got properly translated to Notion).
The argument remains, that calling Objects.requireNonNull to throw a NullPointerException when the very next line will throw the NullPointerException anyway is unnecessary and (arguably) harmful because it produces a _less informative_ exception.

Reply:
Got it, thanks!

I can see it both ways. On one hand, the exception produced with `requireNonNull` is (slightly) less informative. On the other, using `requireNonNull` as a way to enforce method preconditions, and also as a help to IDE's looking for problems, is also useful. Not sure which one wins

Reply:
requireNonNull doesn't help the IDE (it just enforces the `@NonNull` annotation, which _only_ exists to help the IDE and static analysis tools like Sonar).
The IDE (and static analysis tools) will "see" the possible NPE in the code (and highlight it if there is no `@NonNull` annotation present and the inspection is enabled) just as clearly as `Objects.requireNonNull`.

I see the value of `requireNonNull` to move the NPE close to the start of the method, so there is no wasted work and it's more immediately clear to a person reading the code; I don't see any value in `Objects.requireNonNull` if the first 2-3 lines of code will always throw the same exception, however.